app/models/user.rb

```diff
@@ -1,10 +1,36 @@
 class User < ApplicationRecord
-  before_save {self.email = email.downcase}
+  attr_accessor :remember_token
+  before_save{self.email = email.downcase}
   validates :name, presence: true, length: {maximum: Settings.validate.name.maximum}
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: Settings.validate.email.maximum},
     format: {with: VALID_EMAIL_REGEX},
     uniqueness: {case_sensitive: false}
   has_secure_password
   validates :password, presence: true, length: {minimum: Settings.validate.password.minimum}
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```

Comment:
1. I have another question: Can you tell me all of the difference between **update_attribute** and **update_attributes** in active record
2. Use **update_attributes** here

Reply:
The difference is: update_attribute bypasses the validations and update_attribute is used to update record with single attribute:
update_attribute(name, value)
 update_attributes is used to update record with multiple attributes
update_attributes(attributes)